packages/run-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -129,30 +140,43 @@ export const makeVaultManager = (
   );
 
   /**
-   *
-   * @param {[key: string, vaultKit: InnerVault]} record
+   * @param {Iterable<[key: string, vaultKit: InnerVault]>} vaultEntries
    */
-  const liquidateAndRemove = async ([key, vault]) => {
+  const enqueueToLiquidate = vaultEntries => {
     assert(prioritizedVaults);
-    trace('liquidating', vault.getVaultSeat().getProposal());
-
-    try {
-      // Start liquidation (vaultState: LIQUIDATING)
-      await liquidate(
-        zcf,
-        vault,
-        runMint.burnLosses,
-        liquidationStrategy,
-        collateralBrand,
-      );
-
-      await prioritizedVaults.removeVault(key);
-    } catch (e) {
-      // XXX should notify interested parties
-      console.error('liquidateAndRemove failed with', e);
+    for (const [k, v] of vaultEntries) {
+      vaultsToLiquidate.init(k, v);
+      prioritizedVaults.removeVault(k);
     }
   };
 
+  const executeLiquidation = async () => {
+    // Start all promises in parallel
+    // XXX we should have a direct method to map over entries
+    const liquidations = Array.from(vaultsToLiquidate.entries()).map(
+      async ([key, vault]) => {
+        trace('liquidating', vault.getVaultSeat().getProposal());
+
+        try {
+          // Start liquidation (vaultState: LIQUIDATING)
+          await liquidate(
+            zcf,
+            vault,
+            runMint.burnLosses,
+            liquidationStrategy,
+            collateralBrand,
+          );
+
+          vaultsToLiquidate.delete(key);
+        } catch (e) {
+          // XXX should notify interested parties
```

Comment:
If the vault is now in the `liquidating` state, should we try again to remove it from `vaultsToLiquidate`?

Reply:
Good question. My thought was it still needs to be liquidated. This way it'll keep retrying. I don't know of failure modes in which you wouldn't want to retry. I figure we'll come back to that when "notify interested parties" is addressed.